src/js/Aladin.js

```diff
@@ -1437,23 +1445,36 @@ export let Aladin = (function () {
     Aladin.prototype.getViewWCS = function (options) {
         var raDec = this.getRaDec();
         var fov = this.getFov();
-        // TODO: support for other projection methods than SIN
+        var projectionName = this.getProjectionName();
+
+        // get the cootype prefix from the coordinate frame
+        switch (this.getCooFrame()) {
+            case "J2000":
+            case "J2000d":
+                var cooType1 = "RA---";
+                var cooType2 = "DEC--";
+                break;
+            case "Galactic":
+                var cooType1 = "GLON-";
+                var cooType2 = "GLAT-";
+        }
+
         return {
             NAXIS: 2,
             NAXIS1: this.view.width,
             NAXIS2: this.view.height,
-            RADECSYS: 'ICRS',
+            RADECSYS: "ICRS",
             CRPIX1: this.view.width / 2,
             CRPIX2: this.view.height / 2,
             CRVAL1: raDec[0],
             CRVAL2: raDec[1],
-            CTYPE1: 'RA---SIN',
-            CTYPE2: 'DEC--SIN',
+            CTYPE1: cooType1 + projectionName,
+            CTYPE2: cooType2 + projectionName,
             CD1_1: fov[0] / this.view.width,
```

Comment:
this should be fine, *but* I noticed in allsky projections (AIT or Mollweide for instance), when fully zoomed out, *aladin.getFov()* mistakenly returns [180, 180] instead of the expected [360, 180].
This makes the returned WCS erroneous in those cases.

Reply:
Oh that's strange, you suggest we change getFov ? 
Quite a few functions depend on it: 

- `getShareURL`
- `getEmbedCode`
and our `getViewWCS` so (note for later) we'll have to check if they still work

Reply:
Yes, getFov should be fixed.
I haven't checked yet, but wouldn't be surprised if the 180° limitation was there because in v2, we only had SINus projection available.
Other functions calling getFov will also benefit from the fix.

Reply:
Should be fixed. 
Interestingly, there is also a `getFieldOfView` defined on the Rust side in `app.rs`. This one is not exposed to the users and returns a f64 instead of two values.